gym/envs/registration.py

```diff
@@ -680,63 +310,168 @@ def make(id: Literal[
     "Ant-v2"
 ], **kwargs) -> Env[np.ndarray, np.ndarray]: ...
 
-# ----------------------------------------
 
 @overload
 def make(id: str, **kwargs) -> Env: ...
+@overload
+def make(id: EnvSpec, **kwargs) -> Env: ...
+
 # fmt: on
-def make(id: str, **kwargs) -> Env:
-    return registry.make(id, **kwargs)
 
 
-def spec(id: str) -> EnvSpec:
-    return registry.spec(id)
+# Global registry of environments. Meant to be accessed through `register` and `make`
+registry: dict[str, EnvSpec] = dict()
+current_namespace: Optional[str] = None
+
+
+def check_spec_register(spec: EnvSpec):
```

Comment:
This public function lacks a docstring. What it means to "check" spec register?

Reply:
Good catch, I'll add that

Reply:
This should probably be renamed. The intention is that it performs a bunch of checks to make sure that the registration will be valid. It also shouldn't be public